cmake/FindELPA.cmake

```diff
@@ -48,7 +49,7 @@ find_package_handle_standard_args(ELPA DEFAULT_MSG ELPA_LINK_LIBRARIES ELPA_INCL
 
 # Copy the results to the output variables and target.
 if(ELPA_FOUND)
-    set(ELPA_LIBRARY ${ELPA_LINK_LIBRARIES})
+    list(GET ELPA_LINK_LIBRARIES 0 ELPA_LIBRARY)
```

Comment:
What is this line for?

Reply:
To use CMake ELPA::ELPA target without editing ScaLAPACK/LAPACK(and non-existing BLAS) link method.

Reply:
As one return variable of pkg_search_module, ELPA_LINK_LIBRARIES would contain multiple libraries to link against ELPA, as in ${ELPA_ROOT}/lib/pkgconfig/elpa_openmp.pc shows, libelpa_openmp.so as well as dependent libs like scalapack, blas, etc. However, currently ABACUS link against scalapack/openblas through using their full path instead of -L${path} -l${lib}, making -l flags of these libs unusable. This line cuts the dependents off and makes sure only ELPA lib would be linked through this target.
I would say, this is not a good practice, not only cutting dependents off, but also linking dynamic libraries through their FULL path instead of a "dynamic" -L${path} -l${lib} tune. Using line like this only aims to provide a quick fix and avoid editing too much in cmake modules; the remaining part is when to remake ABACUS's linalg libs link modules: as mentioned in #3497, it would be done.

Reply:
A better practice is setting ELPA::ELPA a INTERFACE target and use properties like INTERFACE_LINK_LIBRARIES, INTERFACE_LINK_PATH to manage link flags, that would get CMake link through a -L${path} -l${lib} tune. Currently using INTERFACE style still needs some cut-off to ban dependents link flags like -lscalapack, would you prefer to switch to this style in FindELPA.cmake first?

Reply:
Another solution is, only using pkg_search_module to find ELPA headers with legacy method for libs. A disadvantage is ELPA may contain some dependent libs like rocblas, which would not be reflected through the legacy way.

Reply:
@yizeyi18 Your proposal would be fine, especially for the case that ELPA and ABACUS is built with different math libraries. I didn't realize that `pkg-config` would return all required libraries. Maybe we could discuss the solution in another issue/PR.